tests/system/feature_store/test_feature_store.py

```diff
@@ -3252,6 +3252,328 @@ def test_pandas_stats_include_index(self, index_columns):
+        departments = pd.DataFrame(
+            {
+                "d_id": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+                "name": [
+                    "dept1",
+                    "dept2",
+                    "dept3",
+                    "dept4",
+                    "dept5",
+                    "dept6",
+                    "dept7",
+                    "dept8",
+                    "dept9",
+                    "dept10",
+                ],
+                "manager_id": [i for i in range(10, 20)],
+            }
+        )
+
+        employees = pd.DataFrame(
+            {
+                "id": [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000],
+                "name": [
+                    "name1",
+                    "name2",
+                    "name3",
+                    "name4",
+                    "name5",
+                    "name6",
+                    "name7",
+                    "name8",
+                    "name9",
+                    "name10",
+                ],
+                "department_id": [1, 1, 1, 2, 2, 2, 6, 6, 6, 11],
+            }
+        )
+
+        employees_mini = pd.DataFrame(
+            {
+                "id": [100, 200, 300],
+                "name": [
+                    "name1",
+                    "name2",
+                    "name3",
+                ],
+                "department_id": [1, 1, 1],
+                "class_id": [i for i in range(20, 23)],
+            }
+        )
+
+        classes = pd.DataFrame(
+            {
+                "c_id": [i for i in range(20, 40, 2)],
+                "name": [
+                    "class1",
+                    "class2",
+                    "class3",
+                    "class4",
+                    "class5",
+                    "class6",
+                    "class7",
+                    "class8",
+                    "class9",
+                    "class10",
+                ],
+            }
+        )
+
+        managers = pd.DataFrame(
+            {
+                "m_id": [i for i in range(10, 30, 2)],
+                "name": [
+                    "manager1",
+                    "manager2",
+                    "manager3",
+                    "manager4",
+                    "manager5",
+                    "manager6",
+                    "manager7",
+                    "manager8",
+                    "manager9",
+                    "manager10",
+                ],
+            }
+        )
+
+        result = pd.merge(
+            employees,
+            departments,
+            how=join_type,
+            left_on=["department_id"],
+            right_on=["d_id"],
+            suffixes=("_employees", "_departments"),
+        )
+
+        result_2 = pd.merge(
+            result,
+            managers,
+            how=join_type,
+            left_on=["manager_id"],
+            right_on=["m_id"],
+            suffixes=("_manage", "_"),
+        )
+
+        result_3 = pd.merge(
```

Comment:
what does "employee_mini" and the "class" adds in terms of coverage of the join_type functionality? if more of the same, consider removing - will make the test much simpler.

Reply:
We can now join more when 2 feature_sets so I need more when 2 for testing it.